### PR TITLE
Topic/difference interpolate fix

### PR DESCRIPTION
**Pull request by fvalenza**
related to #278

### CODE REVIEW

File: src/multibody/joint/joint-spherical.hpp [hunk 1]
```diff
@@ -264,9 +264,19 @@ namespace se3
     using JointModelBase<JointModelSpherical>::setIndexes;
     typedef Motion::Vector3 Vector3;
     typedef double Scalar;
+    typedef Eigen::Map<const Motion_t::Quaternion_t> ConstQuaternionMap_t;
 
     JointDataDerived createData() const { return JointDataDerived(); }
 
+    inline void forwardKinematics(Transformation_t & M, ConstQuaternionMap_t & q_joint) const
```
Comment (jcarpent) [inline src/multibody/joint/joint-spherical.hpp]: Use Eigen::MatrixBase<D> instead.

File: src/multibody/joint/joint-spherical.hpp [hunk 2]
```diff
@@ -375,23 +385,16 @@ namespace se3
...
-      const Motion_t::Quaternion_t quat_relatif (quat1*quat0.conjugate());
+      Eigen::VectorXd::ConstFixedSegmentReturnType<NQ>::Type & q_0 = q0.segment<NQ> (idx_q ());
+      Eigen::VectorXd::ConstFixedSegmentReturnType<NQ>::Type & q_1 = q1.segment<NQ> (idx_q ());
+
+      ConstQuaternionMap_t quat0 (q_0.data());
+      ConstQuaternionMap_t quat1 (q_1.data());
+      Transformation_t M0; forwardKinematics(M0, quat0);
+      Transformation_t M1; forwardKinematics(M1, quat1);
+
+      return se3::log3((M0.inverse()*M1).rotation());
```
Comment (jcarpent) [inline src/multibody/joint/joint-spherical.hpp]: `M0.rotation().transpose()*M1.rotation()`to be efficient.

File: unittest/joint-configurations.cpp [hunk 3]
```diff
@@ -151,280 +151,227 @@ BOOST_AUTO_TEST_CASE ( integration_test )
...
-{
-  Model model; buildModel(model);
-  Data data(model);
-
-  std::vector<Eigen::VectorXd> qs1(3);
-  std::vector<Eigen::VectorXd> qs2(3);
-  std::vector<double> us(3);
-  std::vector<Eigen::VectorXd> results(3);
-  std::vector<Eigen::VectorXd> expecteds(3);
+// BOOST_AUTO_TEST_CASE ( interpolation_test )
```
Comment (jcarpent) [inline unittest/joint-configurations.cpp]: Prefer delation instead of just commenting.

**Comment by fvalenza** [comment]
Should I then change the declaration of both forwardKinematics and log3 or only forwardKinematics

**Comment by jcarpent** [comment]
You must take an Eigen::MatrixBase as input and perform the section of the right DoFs inside the FK method.

**Comment by jcarpent** [comment]
Looks good for me. Good job!

**Pull request merged**